cpp/harvest/remid.sh

```diff
@@ -1,47 +1,123 @@
+
+TotalRecord=$(sed -n '/srw:numberOfRecords/{s/.*<srw:numberOfRecords>\(.*\)<\/srw:numberOfRecords>.*/\1/;p}' <<< tmp_total_rec $GET_TOTAL_RECORDS)
+
+rm $GET_TOTAL_RECORDS
+## End Get the total number of records
+
+NumOfIteration=$(( (TotalRecord + RECORDS_PER_CALL - 1) / RECORDS_PER_CALL ))
+
+ColorEcho "++++ Total record found ${TotalRecord}, and it will split into ${NumOfIteration} file(s) ++++"
 
-# maximumRecords is limited to 1000. Since we have about 1300 datasets,
-# we need to download the data in multiple steps and merge them together afterwards
-declare -r RECORDS_PER_CALL=1000
-startRecord=1
-wget "http://sru.hebis.de/sru/DB=2.1?query=pica.abr+%3D+%22REMID%22+or+pica.prv+%3D+%22REMID%22&version=1.1&operation=searchRetrieve&recordSchema=marc21&maximumRecords=${RECORDS_PER_CALL}&startRecord=${startRecord}&recordPacking=xml&sortKeys=LST_Y%2Cpica%2C0%2C%2C" -O $FILE_ORI
+Counter=1
+TempFile=""
+CleanFile=""
+StartRecord=0
 
-# Remove certain tags from non-MARC namespaces so we can use our tools
-grep -vE 'xml-stylesheet|srw:|diag:|:dc|:diag|:xcql|<startRecord|</startRecord|<maximumRecords|</maximumRecords|<sortKeys|</sortKeys|<rob:|</rob:' $FILE_ORI > $FILE_TEMP
+while [ $Counter -le $NumOfIteration ]
+do  
+    ColorEcho "++++ Downloading file ${Counter} of ${NumOfIteration} ++++"
 
-# Broken head+tail need to be removed by hand (including stylesheet and so on)
-# also: insert <collection> root element at start/end.
-sed -i '1 a <collection xmlns="http://www.loc.gov/MARC21/slim" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.loc.gov/standards/marcxml/schema/MARC21slim.xsd">' $FILE_TEMP
+    TempFile="${TEMP_FILE_NAME_PREFIX}_${Counter}.xml"
+    CleanFile="${CLEAN_FILE_NAME_PREFIX}_${Counter}.xml"
 
-echo '</collection>' >> $FILE_TEMP
+    StartRecord=$(( ( Counter - 1) * RECORDS_PER_CALL + 1 ))
+    ColorEcho "Downloading start from record ${StartRecord}"
+    # download some records into the file
+    wget "http://sru.hebis.de/sru/DB=2.1?query=pica.abr+%3D+%22REMID%22+or+pica.prv+%3D+%22REMID%22&version=1.1&operation=searchRetrieve&recordSchema=marc21&maximumRecords=${RECORDS_PER_CALL}&startRecord=${StartRecord}&recordPacking=xml&sortKeys=LST_Y%2Cpica%2C0%2C%2C" -O $TempFile
+
+    ColorEcho "Purging the file... \n\n"
+    # clean the temp file
+    grep -vE '<?xml\ version|xml-stylesheet|srw:|diag:|:dc|:diag|:xcql|<startRecord|</startRecord|<maximumRecords|</maximumRecords|<sortKeys|</sortKeys|<rob:|</rob:' $TempFile > $CleanFile    
+
+    # remove temp file
+    rm $TempFile
+
+    (( Counter++ ))
+done
+
+# merge all clean files
+ColorEcho "++++ Merge all clean files ++++"
+
+Counter=1
+echo '<?xml version="1.0" encoding="UTF-8" ?>' >> $FILE_ORI
+while [ $Counter -le $NumOfIteration ]
```

Comment:
Maybe it would be more efficient to merge this second loop into the first and concat directly after downloading